Mage/src/main/java/mage/abilities/dynamicvalue/DynamicValue.java

```diff
@@ -17,7 +19,39 @@ public interface DynamicValue extends Serializable, Copyable<DynamicValue> {
 
     DynamicValue copy();
 
-    String getMessage();
+    // TODO: this will be removed once all DynamicValue implementations have been converted
+    // WARNING: You must override at least one getMessage in a DynamicValue implementation, otherwise these will loop infinitely.
+    default String getMessage(){
+        return getMessage(ValuePhrasing.LEGACY);
+    }
+
+    /**
+     *
+     * @return A description of what this Dynamic Value represents.
+     * Factor in the Phrasing, which changes the plurality (see {@link mage.constants.ValuePhrasing})
+     * If this value represents something that isn't normally phrased like a discreet count, feel free to ignore the phrasing argument.
+     * For example:
+     *      "the number of creatures you control" or [for each] "creature you control" (discreet count)
+     *      vs
+     *      "the sacrificed creature's power" (non-discreet)
+     */
+    default String getMessage(ValuePhrasing phrasing){
+        // TODO: this will lose its default status once all DynamicValue implementations have been converted
+        return getMessage();
+    }
+
+    /**
+     *
+     * @return A ValueHint with a shortened descriptor of this DynamicValue
+     * example:
+     *      getMessage(EQUAL_TO) -> the number of card types among cards in your graveyard
+     *      getHint() -> ValueHint("Card types in your graveyard", this)
+     * Note: make sure to capitalize the ValueHunt name field
+     */
+    default ValueHint getValueHint(){
+        //TODO: this will become getHint() once all DynamicValue implementations have been converted
```

Comment:
curious what you see as the distinction between getHint and getValueHint?

Reply:
It seems cleaner to me to use `getHint`since many values were already using `getHint`--i saw the name `getValueHint` as a way to avoid colliding with the existing hint getters. I could also just not do this TODO and leave them as `getValueHint` if you still prefer that name

Reply:
Sounds good, thanks for explaining.